src/components/AccountDetails.tsx

```diff
@@ -0,0 +1,111 @@
+import "./AccountDetails.scss"
+
+import React, { ReactElement } from "react"
+import { commify, formatBNToString } from "../utils"
+import Copy from "./Copy"
+import Identicon from "./Identicon"
+import { SUPPORTED_WALLETS } from "../constants"
+import { Zero } from "@ethersproject/constants"
+import { getEtherscanLink } from "../utils/getEtherscanLink"
+import { shortenAddress } from "../utils/shortenAddress"
+import { useActiveWeb3React } from "../hooks"
+import { usePoolTokenBalances } from "../state/wallet/hooks"
+import { useTranslation } from "react-i18next"
+
+interface Props {
+  openOptions: () => void
+}
+
+export default function AccountDetail({ openOptions }: Props): ReactElement {
+  const { t } = useTranslation()
+  const { account, connector } = useActiveWeb3React()
+  const tokenBalances = usePoolTokenBalances()
+  const ethBalanceFormatted = commify(
+    formatBNToString(tokenBalances?.ETH || Zero, 18, 6),
+  )
+
+  const connectorName = Object.keys(SUPPORTED_WALLETS)
+    .filter((k) => SUPPORTED_WALLETS[k].connector === connector)
+    .map((k) => SUPPORTED_WALLETS[k].name)[0]
```

Comment:
can simplify this with [find from lodash](https://lodash.com/docs/4.17.15#find) (which we're already including as a dependency):
```js
import { find } from "lodash"

const connectorName = find(SUPPORTED_WALLETS, ["connector", connector])?.name
```